pkgs/test/lib/src/runner/browser/compilers/dart2js.dart

```diff
@@ -174,7 +174,10 @@ class Dart2JsSupport extends CompilerSupport with JsHtmlWrapper {
   @override
   (Uri, Future<WebSocketChannel>) get webSocket {
     var completer = Completer<WebSocketChannel>.sync();
-    var path = _webSocketHandler.create(webSocketHandler(completer.complete));
+    var path =
+        _webSocketHandler.create(webSocketHandler((WebSocketChannel ws, _) {
```

Comment:
Without the type here, do we get a lint about the implicit cast from dynamic? Or is it an actual error?

In version 3.0 of shelf_web_socket the type will not need to be here, so we probably want to remove it, but I want to make sure we don't break users on 2.0. If that will be the case, we would want a comment here saying not to remove the type. Or, we would want to set the min constraints on shelf_web_socket to >3.0.

Reply:
We get a lint on the webSocketHandler call (`The type of ws can't be inferred; a type must be explicitly provided.`) and an error on the completer line (`The argument type 'dynamic' can't be assigned to the parameter type 'FutureOr<WebSocketChannel>?'. `).

I'll add comments that the type is needed for compatibility with package:shelf_web_socket v2.